src/libmain/shared.cc

```diff
@@ -48,15 +48,21 @@ void printMissing(ref<Store> store, const StorePathSet & willBuild,
     unsigned long long downloadSize, unsigned long long narSize, Verbosity lvl)
 {
     if (!willBuild.empty()) {
-        printMsg(lvl, "these derivations will be built:");
+        printMsg(lvl, fmt("%s will be built:",
+                willBuild.size() == 1
+                    ? "this derivation"
+                    : fmt("these %d derivations", willBuild.size())));
         auto sorted = store->topoSortPaths(willBuild);
         reverse(sorted.begin(), sorted.end());
         for (auto & i : sorted)
             printMsg(lvl, fmt("  %s", store->printStorePath(i)));
     }
 
     if (!willSubstitute.empty()) {
-        printMsg(lvl, fmt("these paths will be fetched (%.2f MiB download, %.2f MiB unpacked):",
+        printMsg(lvl, fmt("%s will be fetched (%.2f MiB download, %.2f MiB unpacked):",
+                willSubstitute.size() == 1
+                    ? "this path"
+                    : fmt("these %d paths",willSubstitute.size()),
```

Comment:
BTW if we ever want to add translations to Nix, then splitting strings in this way is annoying, because in many languages the translation of "will be fetched" depends on the number of the subject. So it's better to write
```
if (willSubstitute.size() == 1)
    printMsg(lvl, fmt("this path will be fetched...", ...);
else
    printMsg(lvl, fmt("these %d paths will be fetched...", ...);
```
This is also easier to read IMHO.

Reply:
That is indeed how I had it initially, but I was unhappy with the duplication this incurred.

Reply:
Fixed